### PR TITLE
Introspection: Add client-side routing

### PR DESCRIPTION
**Pull request by juliettepretot**
Adds client side routing, and placeholder details pages for nodes, handles, channels

# Checklist

- [ ] Pull request affects core Oak functionality (e.g. runtime, SDK, ABI)
  - [ ] I have written tests that cover the code changes.
  - [ ] I have checked that these tests are run by
        [Cloudbuild](/cloudbuild.yaml)
  - [ ] I have updated [documentation](/docs/) accordingly.
  - [ ] I have raised an [issue](https://github.com/project-oak/oak/issues) to
        cover any TODOs and/or unfinished work.
- [ ] Pull request includes prototype/experimental work that is under
      construction.

### CODE REVIEW

File: oak_runtime/introspection_browser_client/components/HandleDetails/index.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  if (node === undefined) {
+    return <p>A node with the ID: {node} does not exist.</p>;
+  }
+
+  const channelHalf = node.abiHandles.get(BigInt(handle));
+
+  if (channelHalf === undefined) {
+    return (
+      <p>
+        The handle {handle} with the does not exist on the node with the ID:{' '}
```
Comment (conradgrobler) [inline oak_runtime/introspection_browser_client/components/HandleDetails/index.tsx:45]: nit: perhaps delete the first instance of "with the"?

File: oak_runtime/introspection_browser_client/components/ChannelDetails/index.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+export default function ChannelDetails({
+  applicationState,
+}: ChannelDetailsProps) {
+  const { channelId } = useParams<ParamTypes>();
+  const channel = applicationState.channels.get(BigInt(channelId));
+
+  if (channel === undefined) {
+    return <p>A channel with the ID: {channel} does not exist.</p>;
```
Comment (conradgrobler) [inline oak_runtime/introspection_browser_client/components/ChannelDetails/index.tsx:36]: nit: should it be {channelId}?

**Review by conradgrobler — APPROVED** [review]

File: oak_runtime/introspection_browser_client/components/HandleDetails/index.tsx (hunk 1 above)
Comment (juliettepretot) [reply oak_runtime/introspection_browser_client/components/HandleDetails/index.tsx:45]: good point! Classic me 😅

File: oak_runtime/introspection_browser_client/components/ChannelDetails/index.tsx (hunk 2 above)
Comment (juliettepretot) [reply oak_runtime/introspection_browser_client/components/ChannelDetails/index.tsx:36]: Yup :))

**Pull request merged**